core/src/main/scala/com/nvidia/spark/rapids/tool/tuning/AutoTuner.scala

```diff
@@ -1181,6 +1181,37 @@ class AutoTuner(
   }
 }
 
+/**
+ * Implementation of the `AutoTuner` designed for the Profiling Tool. This class can be used to
+ * implement the logic to recommend AutoTuner configurations by the Profiling Tool.
+ */
+class ProfilingAutoTuner(
+    clusterProps: ClusterProperties,
+    appInfoProvider: AppSummaryInfoBaseProvider,
+    platform: Platform,
+    driverInfoProvider: DriverLogInfoProvider)
+  extends AutoTuner(clusterProps, appInfoProvider, platform, driverInfoProvider,
+    ProfilingAutoTunerConfigsProvider) {
+
+  /**
+   * Overrides the default recommendation for 'spark.sql.files.maxPartitionBytes'.
+   * Logic:
+   * - If GPU OOM errors occurred in scan stages, halve the maxPartitionBytes.
+   * - Otherwise, calculate using the default implementation (based on input sizes).
+   */
+  override def recommendMaxPartitionBytes(): Unit = {
+    val propKey = "spark.sql.files.maxPartitionBytes"
+    getPropertyValue(propKey) match {
+      case Some(value) if appInfoProvider.hasScanStagesWithGpuOom =>
+        // If GPU OOM errors occurred in scan stages, halve the maxPartitionBytes
+        val recommendedValue = (StringUtils.convertToMB(value) / 2).toString
+        appendRecommendationForMemoryMB(propKey, recommendedValue)
```

Comment:
The problem in this block is that we can only do something about the property only IFF it was already set.
Should we check for `hasScanStagesWithGpuOom` anyway and force a value based on calculations if it is not set?
If it has OOM, the value should be `Minimum(1/2 existing value, AutoTuner calculation)`

Reply:
Refactored the logic to override `calculateMaxPartitionBytes`.
```
Logic:
- First, calculate the recommendation based on input sizes (parent implementation).
- If GPU OOM errors occurred in scan stages,
    - If calculated value is defined, choose the minimum between the calculated value and
      half of the current value.
    - Else, halve the current value.
- Else, use the value from the parent implementation.
```